### ISSUE TITLE
exception when cluster is large

### ISSUE THREAD

**Issue by grayaii** [ExpediaDotCom/c3vis#19]
We're getting the following stack trace in the logs when our cluster is around 40 instances:
```
TypeError: Cannot read property 'PrivateIpAddress' of undefined
    at /Users/alexgray/workspace/allrepos/c3vis_alex/routes/index.js:80:112
    at Array.map (native)
    at /Users/alexgray/workspace/allrepos/c3vis_alex/routes/index.js:79:58
    at /Users/alexgray/workspace/allrepos/c3vis_alex/node_modules/promise/lib/core.js:33:15
    at flush (/Users/alexgray/workspace/allrepos/c3vis_alex/node_modules/asap/asap.js:27:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```
It is 100% reproducible in our environment and I'm not sure what could be the problem.

**Comment by mattcallanan** [comment]
Thanks for reporting this @grayaii. 
We've also seen issues with large clusters and I have a theory it's related to timeouts - e.g. c3vis taking too long to gather its data and respond. Any info you can provide on the timings between the request and error would be helpful. 
Also, what client-side errors do you see?
Thanks

**Comment by xriu** [comment]
For me is happening as well when the instance it's in 'DRAINING' status, the workaround its filter by 'RUNNING' status.

**Comment by grayaii** [comment]
I just built head (which is git hash 5fbab05) and I'm consistently getting throttling errors, which I didn't get when using git hash 055e904.

Here is the output from the logs:

"requestId: '8ce19a1a-0ba3-11e8-b8ba-01d9cf2db74b',",
"time: Wed Feb 07 2018 01:09:30 GMT+0000 (UTC),",
"code: 'ThrottlingException',",
"message: 'Rate exceeded',",
"retryable: true }",
"statusCode: 400,",
"requestId: '8ce19a1a-0ba3-11e8-b8ba-01d9cf2db74b',",
"time: Wed Feb 07 2018 01:09:30 GMT+0000 (UTC),",
"code: 'ThrottlingException',",
"message: 'Rate exceeded',",
"Caught error in getTasksWithTaskDefinitions(): { [ThrottlingException: Rate exceeded]",
null,
"Found 190 tasks",
"Getting Tasks annotated with Task Definitions for cluster ' prod '...",
"/api/instance_summaries_with_tasks: cluster='prod', live=true)",
"[0mGET /api/cluster_names \u001b[32m200 \u001b[0m82.766 ms - 18\u001b[0m",
"[0mGET /javascripts/graph.js \u001b[32m200 \u001b[0m1.711 ms - 12835\u001b[0m",
"[0mGET /javascripts/types.js \u001b[32m200 \u001b[0m0.628 ms - 1156\u001b[0m",
"[0mGET /javascripts/thirdparty/js.cookie.js \u001b[32m200 \u001b[0m1.819 ms - 3241\u001b[0m",
"[0mGET /javascripts/thirdparty/spin.min.js \u001b[32m200 \u001b[0m0.822 ms - 4262\u001b[0m",
"[0mGET /javascripts/thirdparty/bootstrap.min.js \u001b[32m200 \u001b[0m1.217 ms - 36816\u001b[0m",
"[0mGET /javascripts/thirdparty/jquery.min.js \u001b[32m200 \u001b[0m0.943 ms - 84380\u001b[0m",
"[0mGET /javascripts/thirdparty/colorbrewer.js \u001b[32m200 \u001b[0m1.745 ms - 18990\u001b[0m",
"[0mGET /javascripts/thirdparty/d3-context-menu.js \u001b[32m200 \u001b[0m1.528 ms - 2015\u001b[0m",
"[0mGET /javascripts/thirdparty/d3.min.js \u001b[32m200 \u001b[0m1.217 ms - 151145\u001b[0m",
"[0mGET /stylesheets/thirdparty/d3-context-menu.css \u001b[32m200 \u001b[0m0.935 ms - 449\u001b[0m",
"[0mGET /stylesheets/app.css \u001b[32m200 \u001b[0m4.939 ms - 861\u001b[0m",
"[0mGET /stylesheets/thirdparty/bootstrap.min.css \u001b[32m200 \u001b[0m9.481 ms - 122540\u001b[0m",
"[0mGET / \u001b[32m200 \u001b[0m16.939 ms - 7001\u001b[0m",

**Comment by mattcallanan** [comment]
@grayaii the [latest release 0.5.0](https://github.com/ExpediaDotCom/c3vis/releases/tag/0.5.0) addresses the throttling issue. 
You can tweak the values that affect throttling either a) once for all environments (by editing `config/defaults.js`) or b) it per environment you run c3vis in using `$TARGET_ENV` and setting up `config/env/$TARGET_ENV.js` - see https://github.com/ExpediaDotCom/c3vis/blob/master/docs/CONFIGURATION.md.

I'm hoping the other errors reported here are fixed by the 0.5.0 release too @grayaii @xriu .

I'll close this issue for now, but let us know if not fixed and we can address.
